src/ripple/app/ledger/Ledger.cpp

```diff
@@ -914,7 +916,7 @@ Ledger::read (Keylet const& k) const
     if (! value)
         return nullptr;
     auto sle = std::make_shared<SLE>(
-        value->peekSerializer(), value->key());
+        SerialIter{value->data(), value->size()}, value->key());
```

Comment:
I'm not sure if this is kosher, doesn't `SLE` take the `SerialIter` by reference? What's going on here? It will advance the `SerialIter` to reflect the bytes consumed.